File: backend/school-entry-api/src/main/java/de/eshg/schoolentry/api/pdf/Address.java
/*
 * Copyright 2025 cronn GmbH
 * SPDX-License-Identifier: Apache-2.0
 */

package de.eshg.schoolentry.api.pdf;

import java.util.List;

public record Address(
    String name,
    String street,
    String zipCode,
    String city,
    List<String> phoneNumber,
    String url,
    String addressAddition,
    List<String> email) {}

File: packages/lib-portal/src/formatters/address.ts
/**
 * Copyright 2026 cronn GmbH
 * SPDX-License-Identifier: Apache-2.0
 */

import { formatList } from "./list";

export function formatStreetAndHouseNumber(address?: {
  street?: string;
  houseNumber?: string;
}) {
  return formatList([address?.street, address?.houseNumber], " ");
}

export function formatPostalCodeAndCity(address?: {
  postalCode?: string;
  city?: string;
}) {
  return formatList([address?.postalCode, address?.city], " ");
}
